WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -306,14 +321,22 @@ private extension ProductFormViewController {
         defer {
             navigationController?.popViewController(animated: true)
         }
-        guard regularPrice != product.regularPrice ||
-            salePrice != product.salePrice ||
-            dateOnSaleStart != product.dateOnSaleStart ||
-            dateOnSaleEnd != product.dateOnSaleEnd ||
-            taxStatus != product.productTaxStatus ||
-            taxClass?.slug != product.taxClass else {
+
+        var hasChangedData = false
+        if regularPrice != product.regularPrice ||
+        salePrice != product.salePrice ||
+        dateOnSaleStart != product.dateOnSaleStart ||
+        dateOnSaleEnd != product.dateOnSaleEnd ||
+        taxStatus != product.productTaxStatus ||
+        taxClass?.slug != product.taxClass {
+            hasChangedData = true
+        }
```

Comment:
This is probably outside of the norm but I found that it's always better to **prioritize immutability** whenever possible. 🙂 Code is usually easier to reason about if variables are immutable. 

```suggestion
        let hasChangedData: Bool = {
            regularPrice != product.regularPrice ||
                salePrice != product.salePrice ||
                dateOnSaleStart != product.dateOnSaleStart ||
                dateOnSaleEnd != product.dateOnSaleEnd ||
                taxStatus != product.productTaxStatus ||
                taxClass?.slug != product.taxClass
        }()

```

Reply:
Really good observation! ;) I made the change.